src/modules/mapper/generic.mapper.ts

```diff
@@ -0,0 +1,28 @@
+import {Hashmap} from "../DTO/hashmap";
+import {Domain} from "../domains/domain";
+
+export class GenericMapper {
+
+    public static toHashmap(domainModel: Domain): Hashmap {
+        const objectDTO: Hashmap = {};
+
+        const domainProperties = this.getClassPropertyIdentifiers(domainModel);
+
+        domainProperties.forEach(property => {
+            if (domainModel[property]) {
+                objectDTO[property] = domainModel[property];
+            } else {
+                throw new Error(`GenericMapper: domain should implement getter to ${property}`);
+            }
+        })
+
+        return objectDTO;
+    }
+
+    private static getClassPropertyIdentifiers(domainModel: Domain): Array<string> {
```

Comment:
I would set this method in the domain class. So I propose to add this method in the interface implemented by the domains so that any domain should have it. If you think that all the domains will have the same implementation, we can have a domain base class and move it there, so that it is not repeated in every domain.
What do you think @elias480?

Reply:
I think that you say is a better approach.
I'm going to make the changes.

Reply:
My idea was to make a generic method to avoid make the implementation in each domain.

I'm not sure if I understand you,  the first part about the method `getClassPropertyIdentifiers` I think it's better to implement in each domain class. Referent to the second part, it would be good to replace the domain interface by abstract class with the `toHashmap` method?

Reply:
Could you explain a bit more about using an abstract class instead of an interface? Which method/implementation would you set in the abstract class?

Reply:
Thinking about it I found these options:

1. Replace domain interface by domain abstract class. Then our domain classes like Users will extend from Domain class. (big refactor) 
```
export abstract class Domain {
    public abstract toDTO(): Hashmap;
    public abstract whoAmI(): String;
    public toHashmap() {
        // put here the implementation of GenericMapper
    }
}

class User extends Domain {}
```
2. Create new interface with the method `toHashmap` and implement it in each domain class next to the domain interface.
3. Add `toHashmap` method to the current domain interface.

I think `toHashmap` and `toDTO` method are the same, maybe we can leave only one of them.

@sergiogragera @nmanero Can you suggest another option to discuss?
What do you think about the above options?

Reply:
First of all the toDTO and toHashmap methods are different because we can use de toDTO to model the REST response objects hiding attributes such  password for example and the toHashmap method (I would call it toPersistence) we can use it to model the object in the persistence layer (different column names).

On the other hand, I think that a generic implementation is not useful. We need a common interface but we need to have different implementations because each entity will expose different attributes and will model a different hashmap according to the database table for example.